tests/suite/test_app_protect_grpc.py

```diff
@@ -0,0 +1,168 @@
+    delete_ap_policy,
+    delete_ap_logconf,
+)
+from suite.resources_utils import (
+    wait_before_test,
+    create_example_app,
+    wait_until_all_pods_are_ready,
+    create_items_from_yaml,
+    delete_items_from_yaml,
+    delete_common_app,
+    replace_configmap_from_yaml,
+    create_ingress_with_ap_annotations,
+    wait_before_test,
+    get_file_contents,
+)
+from suite.yaml_utils import get_first_ingress_host_from_yaml
+
+
+ap_policies_under_test = ["grpc-block-saygoodbye", "grpc-bloc-sayhello"]
+log_loc = f"/var/log/messages"
+valid_resp_txt = "Hello"
+invalid_resp_text = "The request was rejected. Please consult with your administrator."
+
+class BackendSetup:
+    """
+    Encapsulate the example details.
+
+    Attributes:
+        ingress_host (str):
+    """
+
+    def __init__(self, ingress_host, ssl_port):
+        self.ingress_host = ingress_host
+        self.ssl_port = ssl_port
+
+
+@pytest.fixture(scope="function")
+def backend_setup(request, kube_apis, ingress_controller_endpoint, ingress_controller_prerequisites, test_namespace) -> BackendSetup:
+    """
+    Deploy a simple application and AppProtect manifests.
+
+    :param request: pytest fixture
+    :param kube_apis: client apis
+    :param ingress_controller_endpoint: public endpoint
+    :param test_namespace:
+    :return: BackendSetup
+    """
+    print("------------------------- Replace ConfigMap with HTTP2 -------------------------")
+    replace_configmap_from_yaml(kube_apis.v1,
+                            ingress_controller_prerequisites.config_map['metadata']['name'],
+                            ingress_controller_prerequisites.namespace,
+                            f"{TEST_DATA}/appprotect/grpc/nginx-config.yaml")
+
+    policy = request.param["policy"]
+    print("------------------------- Deploy backend application -------------------------")
+    create_example_app(kube_apis, "grpc", test_namespace)
+    wait_until_all_pods_are_ready(kube_apis.v1, test_namespace)
+
+    print("------------------------- Deploy Secret -----------------------------")
+    src_sec_yaml = f"{TEST_DATA}/appprotect/appprotect-secret.yaml"
+    create_items_from_yaml(kube_apis, src_sec_yaml, test_namespace)
+
+    print("------------------------- Deploy logconf -----------------------------")
+    src_log_yaml = f"{TEST_DATA}/appprotect/logconf.yaml"
+    log_name = create_ap_logconf_from_yaml(kube_apis.custom_objects, src_log_yaml, test_namespace)
+
+    print(f"------------------------- Deploy appolicy: {policy} ---------------------------")
+    src_pol_yaml = f"{TEST_DATA}/appprotect/grpc/{policy}.yaml"
+    pol_name = create_ap_policy_from_yaml(kube_apis.custom_objects, src_pol_yaml, test_namespace)
+
+    print("------------------------- Deploy Syslog -----------------------------")
+    src_syslog_yaml = f"{TEST_DATA}/appprotect/syslog.yaml"
+    create_items_from_yaml(kube_apis, src_syslog_yaml, test_namespace)
+    wait_before_test(40)
```

Comment:
not sure if this wait is necessary here. is it for syslog pods? if that is the case, can we wait until the pod is ready?

Reply:
great catch, I'll add waiting for pod! However there could be an issue at times when the pod is ready but endpoint hasn't been assigned which is why the wait is for.

Reply:
I see. maybe we can wait for that condition (we have 1 endpoint), instead of for the pod being ready?